src/Aspire.Hosting.Redis/RedisBuilderExtensions.cs

```diff
@@ -29,10 +30,21 @@ public static IResourceBuilder<RedisResource> AddRedis(this IDistributedApplicat
         ArgumentNullException.ThrowIfNull(builder);
 
         var redis = new RedisResource(name);
+
+        string? connectionString = null;
+
+        builder.Eventing.Subscribe<ConnectionStringAvailableEvent>(redis, async (@event, ct) =>
+        {
+            connectionString = await redis.GetConnectionStringAsync(ct).ConfigureAwait(false);
```

Comment:
Need to figure out if there is a way to suppress log noise for the health check until its actually useful (or potentially redirect the log check logger noise).

Reply:
What does the log noise look like? This is a downside (and upside?) of using the same DI container for these health checks. We have to figure out where health check logs should show up in the dashboard.

Reply:
This is what it looks like right now:

![image](https://github.com/user-attachments/assets/7553d998-d40b-434e-b3a2-f2ba30eb753a)

(not in the dashboard).

The only way I can think to reduce this noise is to suppress the default health service logging completely and then have the publisher write errors into the per-resource error logs.

Thoughts?

Reply:
> The only way I can think to reduce this noise is to suppress the default health service logging completely and then have the publisher write errors into the per-resource error logs.

Yes! But pushing logs through the resource will cause it to clobber the logs come from the resource.

Reply:
We could push a copy of the error through to each resource which is depending on it, that way the error details are prepended to a resource which isn't running yet. Also its probably the place someone is going to look when they see that they have a resource in a waiting state.

Reply:
Will leave this as is until we've done the dashboard work so we can at least debug why something isn't working. Once we've piped the health checks through into the dashboard as part of https://github.com/dotnet/aspire/issues/5569 then we can suppress this.